src/asm/format.cpp

```diff
@@ -103,6 +114,36 @@ void FormatSpec::finishCharacters() {
 		state = FORMAT_INVALID;
 }
 
+static std::string escapeString(std::string const &str) {
+	std::string escaped;
+	for (char c : str) {
+		// Escape characters that need escaping
+		switch (c) {
+		case '\\':
+		case '"':
+		case '{':
+			escaped += '\\';
+			[[fallthrough]];
+		default:
+			escaped += c;
+			break;
+		case '\n':
+			escaped += "\\n";
+			break;
+		case '\r':
+			escaped += "\\r";
+			break;
+		case '\t':
+			escaped += "\\t";
+			break;
+		case '\0':
+			escaped += "\\0";
+			break;
```

Comment:
The `default` case is handled below the `[[fallthrough]]`, above here. Similarly to the `appendEscapedString` function in lexer.cpp.

Reply:
Ah, I see it now. I'd prefer if `default` was last, if you don't mind?

Reply:
We have three other `[[fallthrough]]`s to `default`, and even more to other `case`s throughout the repo. Since this mirrors the structure of `appendEscapedString`, can I leave it as-is here, and consider refactoring them all separately?

Reply:
I meant moving both the `default` and the case that falls through to it, but, sure, if you'd rather.

Reply:
Fixed in a0982130537f821d4e4ad73dec9d87ec2f59cf22.